core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// manage_assets is a chaincode that contains all the code related to asset management operations (e.g., Lock, Unlock, Claim)
+// and any related utility functions
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"crypto/sha256"
+	"encoding/base64"
+	"time"
+
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+	"github.com/golang/protobuf/proto"
+	log "github.com/sirupsen/logrus"
+	"github.com/hyperledger-labs/weaver-dlt-interoperability/core/network/fabric-interop-cc/contracts/interop/protos-go/common"
+)
+
+type AssetLockValue struct {
+	ContractId	string	`json:"contractId"`
+	Locker		string	`json:"locker"`
+	Recipient	string	`json:"recipient"`
+	Hash		string	`json:"hash"`
+	ExpiryTimeSecs	uint64	`json:"expiryTimeSecs"`
+}
+
+// function to generate contract id as "SHA256" hash based on the asset lock key (which is combination of asset type and asset it)
+func generateAssetLockKeyAndContractId(assetAgreement *common.AssetExchangeAgreement) (string, string) {
+	hasher := sha256.New()
+	assetLockKey := assetAgreement.Type + assetAgreement.Id
```

Comment:
As I mentioned earlier, it's probably useful to add the locker to the key as well, just to disambiguate future assets that may have the same type and id but belong to different owners.